src/main/java/de/qabel/desktop/config/DefaultClientConfiguration.java

```diff
@@ -70,6 +80,37 @@ public void selectIdentity(Identity identity) {
 	}
 
 	@Override
+	public Date getLastDropPoll(Identity identity) {
+
+		Date dropDate = lastDropMap.get(identity.getKeyIdentifier());
+		if (dropDate != null) {
+			return dropDate;
+		}
+		Date firstDropDate = new Date(0L);
+		setLastDropPoll(identity, firstDropDate);
+		return firstDropDate;
+
+	}
+
+	@Override
+	public void setLastDropPoll(Identity identity, Date lastDropPoll) {
+		lastDropMap.put(identity.getKeyIdentifier(), lastDropPoll);
+		setChanged();
+		notifyObservers();
+	}
+
+	@Override
+	public HashMap<String, Date> getLastDropMap() {
+		return lastDropMap;
+	}
+
+	@Override
+	public void setLastDropMap(HashMap<String, Date> lastDropMap) {
+		this.lastDropMap = lastDropMap;
```

Comment:
this is where a null check could be needed. When unserializing a configuration serialized in a previous version (without lastDropMap), this will probably get called with `null`, causing errors later.